helix-view/src/document.rs

```diff
@@ -1141,6 +1142,29 @@ impl Document {
         self.inlay_hints.remove(&view_id);
     }
 
+    /// Returns the id of a view that this doc contains a selection for,
+    /// making sure it is synced with the current changes
+    /// if possible or there are no selections returns current_view
+    /// otherwise uses an arbitrary view
+    pub fn get_synced_view_id(&mut self, current_view: ViewId, tree: &mut Tree) -> ViewId {
+        if self.selections.contains_key(&current_view) {
+            let view = tree.get_mut(current_view);
+            // only need to sync current view if this is not the current doc
+            if view.doc != self.id {
```

Comment:
This is not necessary.  The current view is always sycned. We don't sync when switching buffer either, only when switching Views.

Reply:
The reason for this being here is the document you might be fetching a view id for might not be the document but might contain the current view in its selection, maybe we should only take this path if we are getting for the current document I'll see when I move it too Editor

Reply:
Hmm I am not sure I understand. The currently selected view (whoch should be current_view) never needs to be synchronized. You only need to synchronize views that are not visible. This applys to all documwnts no matter whether they are currently shown in the view.

Reply:
coming back to this the reason why this is needed is if 
1. you have two views ``a`` and ``b``
2. you open the same document (``z``) in them
3. then in ``b`` open a different document (``x``) (this will be view.doc)
4. then go to ``a`` and make a change to ``z`` that won't be synced with b
5. so if you change to ``b`` and write-all, when looking for a view for ''z'' , ``b`` the current_view will work and need to be synced because the change was done in another view and we never synced that change when switching to this view
I hope that makes sense

Reply:
indeed you are right I was misremembering the details